reporters/wdio/TerraWDIOSpecReporter.js

```diff
@@ -0,0 +1,180 @@
+const WDIOSpecReporter = require('wdio-spec-reporter/build/reporter');
+const stripAnsi = require('strip-ansi');
+const fs = require('fs');
+const endOfLine = require('os').EOL;
+const path = require('path');
+const Logger = require('../../scripts/utils/logger');
+
+const LOG_CONTEXT = '[Terra-Toolkit:terra-wdio-spec-reporter]';
+
+class TerraWDIOSpecReporter extends WDIOSpecReporter {
+  constructor(globalConfig, options) {
+    super(globalConfig);
+    this.options = options;
+    this.runners = [];
+    this.resultJsonObject = {
+      startDate: '',
+      type: 'wdio',
+      locale: '',
+      formFactor: '',
+      theme: '',
+      output: [],
+      endDate: '',
+    };
+    this.fileName = '';
+    this.moduleName = '';
+    this.isMonoRepo = false;
+    this.testDirPath = '';
+    this.setResultsDir = this.setResultsDir.bind(this);
+    this.hasReportDir = this.hasReportDir.bind(this);
+    this.setTestModule = this.setTestModule.bind(this);
+    this.hasMonoRepo = this.hasMonoRepo.bind(this);
+    this.monoRepoPrintSuitesSummary = this.monoRepoPrintSuitesSummary.bind(this);
+    this.setTestDirPath = this.setTestDirPath.bind(this);
+    this.hasMonoRepo();
+    this.setTestDirPath();
+    this.setResultsDir(options);
+    this.hasReportDir();
```

Comment:
Maybe I'm less familar with JS classed, but should this be like:
```suggestion
    this.isMonoRepo = hasReportDir();
```